### PR TITLE
Dict fix

### PR DESCRIPTION
**Pull request by CurtHagenlocher**
### Fixes

Issue #400 

### Description

Fixes issue #400 by decoding both dictionary-encoded and plain-encoded data into the column chunk array as the pages are read instead of waiting until the end and picking either one or the other.

- [x] I have included unit tests validating this fix.
- [x] I have updated markdown documentation where required.
- [x] I understand that successful approval of my pull request requires reproducible tests as per [Contribution Guideline](https://github.com/elastacloud/parquet-dotnet/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: src/Parquet/File/DataColumnReader.cs [hunk 1]
```diff
@@ -214,34 +209,55 @@ private int ReadLevels(BinaryReader reader, int maxLevel, int[] dest, int offset
...
+               _dataTypeHandler.MergeDictionary(cd.dictionary, cd.indexes, cd.values, cd.valuesOffset, indexCount);
+               cd.valuesOffset += indexCount;
                break;
 
             default:
                throw new ParquetException($"encoding {encoding} is not supported.");
          }
       }
 
+      private int[] InitIndexes(int totalValues, int currOffset)
```
Comment (CurtHagenlocher) [inline src/Parquet/File/DataColumnReader.cs]: Oops, forgot to remove these even though the current iteration doesn't use them.

**Comment by aloneguid** [comment]
Should be ready to merge once conflict is fixed

**Comment by CurtHagenlocher** [comment]
Conflict fixed.

**Comment by aloneguid** [comment]
Thank you

**Pull request merged**